### PR TITLE
Better streaming

### PR DESCRIPTION
**Pull request by ml31415**
- Stream reading improved with respect to Sans-IO
- pre-commit updated
- mypy exceptions more fine-grained
- Drop py3.9 support

### CODE REVIEW

**Comment by limx0** [comment]
LGTM, thanks @ml31415! I think we can also bump python to be >=3.10 now if you'd like? 

I'll merge + fire off a release when this is ready to go.

**Comment by ml31415** [comment]
@limx0 Guess it's good now. Let's get a release out. Btw. would be great, if you could add me as a maintainer. Some redundancy never hurts.

**Comment by ml31415** [comment]
Alright, good to go now. Still a bit undertested, but anyways, a good leap forward.

**Review by limx0 — APPROVED** [review]

**Pull request merged**
